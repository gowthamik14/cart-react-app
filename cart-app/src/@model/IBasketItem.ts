import IProduct from './IProduct';

export default interface IBasketItem extends IProduct {
  quantity: number;
}
