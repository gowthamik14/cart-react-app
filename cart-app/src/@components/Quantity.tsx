import React from 'react';
import IBasketItem from '@model/IBasketItem';


interface IQuantityProp {
    item : IBasketItem,
    increaseQuantity : (basketItem: IBasketItem ) => void
    decreaseQuantity :  (basketItem: IBasketItem ) => void
}

const Quantity : React.FC<IQuantityProp> = ({item, increaseQuantity, decreaseQuantity}) => {
return (
<span>
<button
  className="ml-4 p-2 bg-blue-400 text-white rounded-md focus:outline-none focus:ring-2 ring-blue-200 ring-offset-2"
  onClick={() => increaseQuantity({...item})}
>
  +
</button>
<span className="ml-4 p-2 w-10 text-center text-blue rounded-md">
  {item.quantity}
</span>
<button
  className="ml-4 p-2  bg-white-400 text-blue rounded-md border-2 border-blue-400 focus:outline-blue focus:ring-2 ring-blue-200 ring-offset-2"
  onClick={() => decreaseQuantity({...item})}
>
  -
</button>
</span>
);
}

export default Quantity;
