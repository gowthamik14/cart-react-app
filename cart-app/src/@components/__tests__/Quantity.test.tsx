import React from 'react';
import { render, screen, act } from '@testing-library/react';
import Quantity from '@components/Quantity';
import user from '@testing-library/user-event';
import IBasketItem from '@model/IBasketItem';

test('renders learn react link', () => {
  const item: IBasketItem = {
    name: 'cheese',
    price: 20,
    quantity: 3,
    id: 1,
  };

  const increaseQuantity = jest.fn();
  const decreaseQuantity = jest.fn();
  const { container } = render(
    <Quantity
      item={item}
      increaseQuantity={increaseQuantity}
      decreaseQuantity={decreaseQuantity}
    />
  );

  expect(container).toMatchSnapshot();
  act(() => user.click(screen.getByRole('button', { name: '+' })));
  expect(increaseQuantity).toHaveBeenCalledTimes(1);
  act(() => user.click(screen.getByRole('button', { name: '-' })));
  expect(decreaseQuantity).toHaveBeenCalledTimes(1);
});
