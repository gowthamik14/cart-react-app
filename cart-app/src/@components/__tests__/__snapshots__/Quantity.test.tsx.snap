// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <span>
    <button
      class="ml-4 p-2 bg-blue-400 text-white rounded-md focus:outline-none focus:ring-2 ring-blue-200 ring-offset-2"
    >
      +
    </button>
    <span
      class="ml-4 p-2 w-10 text-center text-blue rounded-md"
    >
      3
    </span>
    <button
      class="ml-4 p-2  bg-white-400 text-blue rounded-md border-2 border-blue-400 focus:outline-blue focus:ring-2 ring-blue-200 ring-offset-2"
    >
      -
    </button>
  </span>
</div>
`;
